ledgerwallet/client.py

```diff
@@ -301,22 +302,32 @@ def _load_segment(self, hex_file: IntelHex, segment):
             self._load_chunk(hex_file, segment, offset)
 
     def install_app(self, app_manifest: AppManifest):
-        hex_file = IntelHex(app_manifest.get_binary())
+        version_info = self.get_version_info()
+        device = str(version_info.target_id)
+
+        if isinstance(app_manifest, AppManifestToml):
```

Comment:
This code should not rely on lower-level `AppManifestToml` specifics. As it expects an `AppManifest`, this later should hold inner capabilities managing this case internaly.
We can think of an `AppManifest.assert_compatible_device` method which would be implemented in both `AppManifestJson` and `AppManifestToml` as:

```python
class AppManifestJson(AppManifest):
    def assert_compatible_device(self, device: str): 
        pass

class AppManifestToml(AppManifest):
    def assert_compatible_device(self, device: str):
        if not "binary" in self.dic.get(device, list()):
            raise ValueError(
                "TOML manifest has no installation information about the current"
                f" device '{device}'"
            )
```